### PR TITLE
Fix detection of hostname to use lowercase

### PR DESCRIPTION
**Pull request by caseydavenport**
## Todos
- [ ] Tests
- [ ] Documentation
- [ ] Release note

## Release Note

```release-note
None required
```

### CODE REVIEW

**Comment by gunjan5** [comment]
@caseydavenport seems like the linter is linting the vendor directory?

**Comment by caseydavenport** [comment]
hmmm

**Comment by caseydavenport** [comment]
@gunjan5 not sure what was going on... I've pinned the offending package for now, but we're passing `--vendor` so not sure why it was even checking it...

File: glide.yaml [hunk 1]
```diff
@@ -41,3 +41,5 @@ import:
   - fv.containers
 - package: github.com/coreos/etcd
   version: v3.2.7
+- package: golang.org/x/crypto
+  version: 1351f936d976c60a0a48d728281922cf63eafb8d
```
Comment (gunjan5) [inline glide.yaml]: I think this version causes issues with calicoctl (especially /x/sys windows package) but should be fine in CNI. This pin doesn't seem relevant to this PR, was it conflicting with glide up?

Comment (gunjan5) [reply glide.yaml]: Oh I see your comment about --vendor now.
